axiom/nr_php_packages.h

```diff
@@ -10,20 +10,43 @@
 #include "util_random.h"
 #include "util_vector.h"
 #include "util_hashmap.h"
+#include "util_strings.h"
 
 #define PHP_PACKAGE_VERSION_UNKNOWN " "
 
+typedef enum {
+  NR_PHP_PACKAGE_SOURCE_LEGACY,
+  NR_PHP_PACKAGE_SOURCE_COMPOSER
+} nr_php_package_source_priority_t;
+
 typedef struct _nr_php_package_t {
   char* package_name;
   char* package_version;
+  nr_php_package_source_priority_t source_priority;
```

Comment:
`source_priority` decides if `package_version` gets overwritten when a package with the same `package_name` is added to `nr_php_packages_t.data` again - see implementation of `nr_php_packages_add_package`.